packages/cart-sdk/hooks/use-checkout.ts

```diff
@@ -0,0 +1,270 @@
+import { useAppContext } from '@ifixit/app';
+import { useAuthenticatedUser } from '@ifixit/auth-sdk';
+import { assertNever, isError } from '@ifixit/helpers';
+import { useIFixitApiClient } from '@ifixit/ifixit-api-client';
+import { useShopifyStorefrontClient } from '@ifixit/shopify-storefront-client';
+import * as React from 'react';
+import { CartError } from '../utils';
+import { useCart } from './use-cart';
+
+const LOCAL_CHECKOUT_STORAGE_KEY = 'checkout';
+
+type UseCheckout = CheckoutState & {
+   redirectToCheckout: RedirectToCheckoutFn;
+   reset(): void;
+};
+
+interface CheckoutState {
+   error: CartError | null;
+   isRedirecting: boolean;
+}
+
+interface RedirectToCheckoutFn {
+   (): Promise<void>;
+}
+
+export function useCheckout(): UseCheckout {
+   const cart = useCart();
+   const user = useAuthenticatedUser();
+   const standardCheckout = useStandardCheckout();
+   const draftOrderCheckout = useDraftOrderCheckout();
+   const [state, dispatch] = React.useReducer(reducer, {
+      error: null,
+      isRedirecting: false,
+   });
+   return {
+      ...state,
+      redirectToCheckout: async () => {
+         if (cart.data == null || !cart.data.hasItemsInCart) {
+            dispatch({
+               type: ActionType.FailedCheckout,
+               error: CartError.EmptyCart,
+            });
+         } else {
+            dispatch({ type: ActionType.StartCheckout });
+            let url: string;
+            try {
+               if (user.data?.is_pro) {
+                  url = await draftOrderCheckout();
+               } else {
+                  url = await standardCheckout();
+               }
+               window.location.href = url;
+            } catch (error) {
+               if (isError(error)) {
+                  console.error(error.message);
+               }
+               dispatch({
+                  type: ActionType.FailedCheckout,
+                  error: CartError.UnknownError,
+               });
+            }
+         }
+      },
+      reset: () => {
+         dispatch({ type: ActionType.ResetCheckout });
+      },
```

Comment:
Would we want to do something like `resetLocalCheckout()` here too?

Reply:
Good question 🤔 This initially was meant mostly to dismiss che checkout error message. When an error occurs though it might be a good behaviour to clean up local cache, because maybe that's what's causing the error. What do you think? Should we clear the local checkout too?

Reply:
Hm I don't have a strong opinion. Thinking about it some more, with the way you've implemented `useStandardCheckout()` we're probably fine leaving this as is. If the local checkout is stale, it looks like we'll end up creating a new checkout anyways. Doesn't seem like a bad thing to keep the local checkout around and use it if it is still valid 🤷🏻